File: java/eliuds-eggs/src/main/java/EliudsEggs.java
public class EliudsEggs {
  public int eggCount(int number) {
    // return Integer.bitCount(number);
    // return (int) Integer.toBinaryString(number).codePoints().filter(x -> x == 49).count();
    int count = 0;
    while (number > 0) {
      if (number % 2 == 1) {
        count++;
      }
      number >>= 1;
    }
    var z = Math.PI;
    return count;
  }
}

File: elixir/eliuds-eggs/lib/eliuds_eggs.ex
defmodule EliudsEggs do
  @moduledoc false

  @doc """
  Given the number, count the number of eggs.
  """
  @spec egg_count(number :: integer()) :: non_neg_integer()
  def egg_count(0), do: 0
  def egg_count(n), do: 1 + egg_count(Bitwise.&&&(n, n - 1))
end
